default-fields/0000-default-fields.md

```diff
@@ -28,9 +233,29 @@ Why should we *not* do this?
 # Alternatives
 [alternatives]: #alternatives
 
-What other designs have been considered? What is the impact of not doing this?
+Just stick with builders and constructor functions.
 
 # Unresolved questions
 [unresolved]: #unresolved-questions
 
-What parts of the design are still TBD?
+## `const`
+
+Is `const` the right feature to base off?
+It means field values are just as constrained as `const` is now,
+but also prevents struct definitions from becomes bloated by complex logic that
+maybe belongs in a builder instead.
```

Comment:
Is there anywhere I can get a sense of how struct literals work now, so I've got some context of how this should be implemented?

The simplest implementation would be to treat struct initialisation as sugar for the 'real' initialiser, where defaults for missing fields are spliced in.

In this case, assuming no inlining, you aren't paying for a function call. Because this is opaque to the caller, with field defaults there might be other optimisations the compiler could do to make it more efficient. I'm not sure what those could be yet though.